File: map/lab7/lab7/Repository/IRepository.cs
﻿using System.Collections.Generic;

namespace lab7.Repository
{
    public interface IRepository<T, ID>
    {
        T Save(T t);
        Dictionary<ID, T> GetAll();
        bool Delete(ID id);
        long Size();
        T GetOne(ID id);
        T Update(T t);
    }
}

File: map/24ianpracticeexam/src/repository/IRepository.java
package repository;

import model.HasId;
import model.ValidationException;

public interface IRepository<T extends HasId, ID> {
    T save(T t) throws ValidationException;
    T remove(ID id);
    T update(T t) throws ValidationException;
    T getOne(ID id);
    Iterable<T> getAll();
}
